docs/source/metrics.rst

```diff
@@ -137,6 +137,56 @@ This metrics API is independent of PyTorch Lightning. Metrics can directly be us
     To change this, after initializing the metric, the method ``.persistent(mode)`` can
     be used to enable (``mode=True``) or disable (``mode=False``) this behaviour.
 
+*******************
+Metrics and devices
+*******************
+
+Metrics are simple subclasses of ``torch.nn.Module`` and their metric states behave
+similar to buffers and parameters of modules. This means that metrics states should
+be moved to the same device as the input of the metric:
+
+.. code-block:: python
+
+    import torch
+    from pytorch_lightning.metrics import Accuracy
+
+    target = torch.tensor([1, 1, 0, 0], device=torch.device("cuda", 0))
+    preds = torch.tensor([0, 1, 0, 0], device=torch.device("cuda", 0))
+
+    # Metric states are always initialized on cpu, and needs to be moved to
+    # the correct device
+    confmat = Accuracy(num_classes=2).to(torch.device("cuda", 0))
+    out = confmat(preds, target)
+    print(out.device) # cuda:0
+
+However, when **properly defined** inside a ``LightningModule``, lightning will
+automatically move the metrics to the same device as the data. Being **properly defined**
+means that the metric is correctly identified as a child module of the model 
+(check ``.children()`` attribute of the model). Therefore, metrics cannot be placed 
+in native python ``list`` and ``dict``, as they will not be correctly identified
+as child modules. Instead of ``list`` use ``torch.nn.ModuleList`` and instead of
```

Comment:
```suggestion
as child modules. Instead of ``list`` use :class:`torch.nn.ModuleList` and instead of
```
btw we can directly link torch docs. Hope I didn't make a typo

Reply:
just missed `~`

Reply:
it's optional, it depends if you want to show short or long name :)